src/Command/Helpers/AmplitudeClientTrait.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Acquia\Console\ContentHub\Command\Helpers;
+
+use Acquia\Console\Acsf\Platform\ACSFPlatform;
+use Acquia\Console\Cloud\Platform\AcquiaCloudPlatform;
+use Acquia\Console\ContentHub\Client\AmplitudeClient;
+use Acquia\Console\ContentHub\ContentHubConsoleEvents;
+use Acquia\Console\ContentHub\Event\ServiceClientUuidEvent;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Trait AmplitudeClientTrait.
+ *
+ * Used for initializing Amplitude Client and common methods.
+ *
+ * @package Acquia\Console\ContentHub\Command\Helpers
+ */
+trait AmplitudeClientTrait {
+
+  /**
+   * User details for amplitude tracking.
+   *
+   * @var array
+   */
+  protected $userDetails = [];
+
+  /**
+   * Amplitude Client to log events.
+   *
+   * @var \Acquia\Console\ContentHub\Client\AmplitudeClient
+   */
+  protected $amplitudeClient;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function initializeAmplitudeClient(OutputInterface $output): void {
+    if (empty($this->amplitudeClient)) {
+      $platform = $this->getPlatform('source');
+      $client_origin_uuid = $platform->get(self::SERVICE_UUID_KEY);
+      if (!$client_origin_uuid) {
+        $event = new ServiceClientUuidEvent($platform, $output);
+        $this->dispatcher->dispatch(ContentHubConsoleEvents::GET_SERVICE_CLIENT_UUID, $event);
+        $client_origin_uuid = $event->getClientServiceUuid();
+        if (empty($client_origin_uuid)) {
+          throw new \Exception(sprintf('Service Subscription UUID missing.'));
+        }
+        $platform->set(self::SERVICE_UUID_KEY, $client_origin_uuid);
```

Comment:
If I understand well this is the CH subscription id. Which defines the CH api key and secret key... etc. It wont change. My question here is, wouldn't it be useful to save this value during platform creation? @abarriosr @amangrover90 @kaynenh  If we want it there we should create a follow-up ticket for it!

Reply:
And all in all it would simplify this nested if statements here.

Reply:
We can do it in Platform Creation but it's possible(for Multisites and ACSF) that some of the sites might not be having CH installed like what happened for one Customer as @abarriosr discussed. We would have to refine this anyway when we implement that functionality to filter the sites. At that time, either we can do this in Platform Creation or we can just leave it here.

Reply:
I think its a good idea to revisit saving the value on platform creation when there is a need to track more commands and when the story about restricting the list of sites that have content hub installed is completed. Until then its probably ok to have it this way for now so we can complete this ticket. We can proceed with this for now.

Reply:
Merging.